src/openlrr/game/object/ObjectRecall.cpp

```diff
@@ -22,27 +25,163 @@ LegoRR::ObjectRecall_Globs & LegoRR::objectRecallGlobs = *(LegoRR::ObjectRecall_
 #pragma region Functions
 
 // <LegoRR.exe @00459450>
-//void __cdecl LegoRR::ObjectRecall_StoreMiniFigure(const LegoObject* liveObj);
+void __cdecl LegoRR::ObjectRecall_StoreMiniFigure(const LegoObject* liveObj)
+{
+
+	//Check type is correct, ability flags are set, and name is custom
+	if ((liveObj->type == LegoObject_MiniFigure) &&
+		(((liveObj->abilityFlags != ABILITY_FLAG_NONE || (liveObj->objLevel != 0)) || (liveObj->customName != nullptr)))) {
+		if (objectRecallGlobs.recallUsed == objectRecallGlobs.recallCapacity) {
+			/* More space is needed, allocate 10 extra slots */
+			//FC, should this be OBJECT_MAXRECORDOBJECTS?
+			objectRecallGlobs.recallCapacity = objectRecallGlobs.recallCapacity + 10;
+			objectRecallGlobs.recallList = (ObjectRecallEntry*) Gods98::Mem_ReAlloc(objectRecallGlobs.recallList,objectRecallGlobs.recallCapacity * sizeof(ObjectRecallEntry));
+		}
+		//Grab latest object in list, set flags and strings
+		objectRecallGlobs.recallList[objectRecallGlobs.recallUsed].abilityFlags = liveObj->abilityFlags;
+		objectRecallGlobs.recallList[objectRecallGlobs.recallUsed].level = liveObj->objLevel;
+		std::memset(objectRecallGlobs.recallList[objectRecallGlobs.recallUsed].customName, 0, OBJECT_CUSTOMNAMELENGTH);
+		if (liveObj->customName != nullptr) {
+			std::strncpy(objectRecallGlobs.recallList[objectRecallGlobs.recallUsed].customName, liveObj->customName, OBJECT_MAXCUSTOMNAMECHARS);
+		}
+
+		//Move to next used object
+		objectRecallGlobs.recallUsed++;
+	}
+	return;
+}
 
 // <LegoRR.exe @00459500>
-//bool32 __cdecl LegoRR::ObjectRecall_RecallMiniFigure(LegoObject* liveObj);
+bool32 __cdecl LegoRR::ObjectRecall_RecallMiniFigure(LegoObject* liveObj)
+{
+	if ((liveObj->type == LegoObject_MiniFigure) && (objectRecallGlobs.recallUsed != 0)) {
+		unsigned int prevUsedIndex = objectRecallGlobs.recallUsed - 1;
```

Comment:
The `recallUsed` count needs to be decremented. The `prevUsedIndex` you store is still correct though.
```cpp
objectRecallGlobs.recallUsed--;
```
You can also just use `objectRecallGlobs.recallUsed` after the decrement instead of using `prevUsedIndex`.